sieves/tasks/predictive/ner/bridges.py

```diff
@@ -0,0 +1,493 @@
+        LiteralType = Literal[*entity_types]  # type: ignore[valid-type]
+
+        class EntityWithContext(pydantic.BaseModel):
+            text: str
+            context: str
+            entity_type: LiteralType
+
+        class Prediction(pydantic.BaseModel):
+            entities: list[EntityWithContext] = []
+
+        if self.prompt_signature_description:
+            Prediction.__doc__ = jinja2.Template(self.prompt_signature_description).render()
+
+        return Prediction
+
+    def consolidate(
+        self, results: Iterable[pydantic.BaseModel], docs_offsets: list[tuple[int, int]]
+    ) -> Iterable[pydantic.BaseModel]:
+        results = list(results)
+        prediction_class = self._prompt_signature  # Get the class, not an instance
+
+        # Process each document (which may consist of multiple chunks)
+        for doc_offset in docs_offsets:
+            doc_results = results[doc_offset[0] : doc_offset[1]]
+
+            # Combine all entities from all chunks
+            all_entities: list[dict[str, Any]] = []
+
+            # Process each chunk for this document
+            for chunk_result in doc_results:
+                if not hasattr(chunk_result, "entities") or not chunk_result.entities:
+                    continue
+
+                # Process entities in this chunk
+                for entity in chunk_result.entities:
+                    # We just need to combine all entities from all chunks
+                    all_entities.append(entity)
+
+            # Create a consolidated result for this document - instantiate the class with entities
+            yield prediction_class(entities=all_entities)
+
+
+class OutlinesNER(PydanticBasedNER[outlines_.InferenceMode]):
+    @property
+    def inference_mode(self) -> outlines_.InferenceMode:
+        return outlines_.InferenceMode.json
+
+    @cached_property
+    def prompt_signature(self) -> type[pydantic.BaseModel]:
+        return self._prompt_signature
+
+
+class OllamaNER(PydanticBasedNER[ollama_.InferenceMode]):
+    @property
+    def inference_mode(self) -> ollama_.InferenceMode:
+        return ollama_.InferenceMode.chat
+
+    @property
+    def prompt_signature(self) -> type[pydantic.BaseModel]:
+        return self._prompt_signature
+
+
+class LangChainNER(PydanticBasedNER[langchain_.InferenceMode]):
+    @property
+    def inference_mode(self) -> langchain_.InferenceMode:
+        return langchain_.InferenceMode.structured_output
+
+    @property
+    def prompt_signature(self) -> type[pydantic.BaseModel]:
+        return self._prompt_signature
+
+
+class InstructorNER(PydanticBasedNER[instructor_.InferenceMode]):
+    @property
+    def inference_mode(self) -> instructor_.InferenceMode:
+        return instructor_.InferenceMode.chat
+
+    @property
+    def prompt_signature(self) -> type[pydantic.BaseModel]:
+        return self._prompt_signature
+
+
+class GliXNER(NERBridge[list[str], glix_.Result, glix_.InferenceMode]):
```

Comment:
I have neglected to tell you about [`GliXBridge`](https://github.com/MantisAI/sieves/blob/main/sieves/tasks/predictive/bridges.py#L105) - all the GliNER-related stuff can be done in one single unified bridge. Instead of writing a dedicated NER bridge, adjust `GliXBridge` to support NER.

Reply:
Hm, good point. I'll think about how to refactor this at a later point. Let's go ahead with the bridge as is for this PR.